### PR TITLE
[SDKv2] Add secp256k1 account authenticator

### PR DESCRIPTION
**Pull request by 0xjinn**
### Description

Adds the AccountAuthenticator for secp256k1

### Test Plan

pnpm fmt & pnpm test

### CODE REVIEW

**Comment by 0xjinn** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10410** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10410" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10410?utm_source=stack-comment).

File: ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts [hunk 1]
```diff
@@ -82,3 +83,34 @@ export class AccountAuthenticatorMultiEd25519 extends AccountAuthenticator {
...
+  public readonly signature: Secp256k1Signature;
+
+  constructor(public_key: Secp256k1PublicKey, signature: Secp256k1Signature) {
+    super();
+    this.public_key = public_key;
+    this.signature = signature;
+  }
+
+  /**
+   * Transaction authenticator Secp256k1 for a single signer transaction
```
Comment (xbtmatt) [inline ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts:99]:
```suggestion
   * A Secp256k1 AccountAuthenticator for a single signer
```

or something like this- technically this isn't the `TransactionAuthenticator`, it's the `AccountAuthenticator`

File: ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts [hunk 2]
```diff
@@ -82,3 +83,34 @@ export class AccountAuthenticatorMultiEd25519 extends AccountAuthenticator {
...
+   *
+   * @param public_key Account's Secp256k1 public key.
```
Comment (xbtmatt) [inline ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts:101]:
```suggestion
   * @param public_key A Secp256k1 public key
```

File: ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts [hunk 3]
```diff
@@ -82,3 +83,34 @@ export class AccountAuthenticatorMultiEd25519 extends AccountAuthenticator {
...
+   * @param signature Account's Secp256k1 signature
```
Comment (xbtmatt) [inline ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts:102]:
```suggestion
   * @param signature A Secp256k1 signature
```

**Review by xbtmatt — CHANGES_REQUESTED** [review]
Need to add this to the `AccountAuthenticator` `deserialize` function, other than that looks good

```typescript
      case AccountAuthenticatorVariant.AccountAuthenticatorSecp256k1:
        return AccountAuthenticatorSecp256k1.load(deserializer);
```

**Comment by 0xjinn** [comment]
> Need to add this to the `AccountAuthenticator` `deserialize` function, other than that looks good
> 
> ```ts
>       case AccountAuthenticatorVariant.AccountAuthenticatorSecp256k1:
>         return AccountAuthenticatorSecp256k1.load(deserializer);
> ```

Ah good catch!

File: ecosystem/typescript/sdk_v2/src/types/index.ts [hunk 4]
```diff
@@ -60,6 +60,7 @@ export enum TransactionAuthenticatorVariant {
 export enum AccountAuthenticatorVariant {
   AccountAuthenticatorEd25519 = 0,
   AccountAuthenticatorMultiEd25519 = 1,
+  AccountAuthenticatorSecp256k1 = 2,
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/types/index.ts:63]: lets shorten these?
```
export enum AccountAuthenticatorVariant {
  Ed25519 = 0,
  MultiEd25519 = 1,
  Secp256k1 = 2,
}
```

File: ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts [hunk 5]
```diff
@@ -82,3 +83,34 @@ export class AccountAuthenticatorMultiEd25519 extends AccountAuthenticator {
     return new AccountAuthenticatorMultiEd25519(public_key, signature);
   }
 }
+
+export class AccountAuthenticatorSecp256k1 extends AccountAuthenticator {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts:87]: need to add to the main `AccountAuthenticator` class https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk_v2/src/transactions/authenticator/account.ts#L11

**Review by xbtmatt — APPROVED** [review]
LGTM!

**Review by 0xmaayan — APPROVED** [review]

**Pull request merged**